### DISCUSSION TITLE
Issues with buttons not responding correctly

### DISCUSSION THREAD

**Discussion by rgregg** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#451] [Q&A]
I have the PCB_V2 wired with the V2b with the blue LLC config and everything has been working perfectly from the web interface and MQTT/home assistant. However I've been having issues interacting with the buttons on the pump itself.

For example: when the control panel is locked, I can't seem to unlock it by holding the lock button. Via MQTT I can see that the LOCK button is pressed, but the control panel never moves to be unlocked, so it appears none of the other buttons work. However, via MQTT as well I can see the other buttons are being received, but I frequently see the wrong button showing up. For example, I'll hold the LOCK button, but via MQTT I see the current button being LOCK, then POWER, then UP, then back to LOCK. Similar events occur when holding other buttons down.

This morning when it was raining, the tub was going crazy as the water on the screen was apparently activating various buttons causing all sorts of actions (bubbles on/off, pump on/off, timer function, etc). I was able to tilt the screen up and wipe the water off and that calmed things down, but I'd like to get the issue under control so I can use the buttons when someone is in the tub.

I was trying to figure out if there is a way to remotely debug what is occuring, but it looks like the serial logger only works when the device is directly attached to the computer. Any other ways to diagnose what might be happening on the panel communication?

I also didn't see an MQTT command to lock/unlock the control panel. Is this possible?

Thanks for any advice/tips.

**Comment by visualapproach** [comment]
Sounds like unstable communication with the display. Unless if it behaves like that without the ESP in which case it's obviously a faulty display*. Assuming it's not, and you have confirmed good solders and cables - try replacing LLC with a red one or run without any LLC at all. You would need to move the data wires to the NodeMCU ofc if choosing the latter.
There is no unlock feature via MQTT. It's just an annoyance so every time you send a command, the ESP will power up and unlock the pump automatically if necessary.

*Version 4 should be able to connect a replacement display from another model with the old pump. At least that's my intention. Not tested.

**Comment by Bishbosh02** [comment]
I get this issue if I have the wrong model selected 
![Models2](https://user-images.githubusercontent.com/9953147/222952494-7291a0ac-c139-4242-999d-94df612ac550.JPG)
